root-packages/runc/build.sh

```diff
@@ -6,17 +6,62 @@ TERMUX_PKG_VERSION="1.1.14"
 TERMUX_PKG_SRCURL=https://github.com/opencontainers/runc/archive/v${TERMUX_PKG_VERSION}.tar.gz
 TERMUX_PKG_SHA256=563cf57c38d2e7149234dbe6f63ca0751eb55ef8f586ed12a543dedc1aceba68
 TERMUX_PKG_AUTO_UPDATE=true
-TERMUX_PKG_DEPENDS="libseccomp"
+TERMUX_PKG_BUILD_DEPENDS="libseccomp-static"
 
 termux_step_make() {
+	cat > fakes.c << EOF
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/system_properties.h>
+
+int __android_log_vprint(int prio, const char* tag, const char* fmt, va_list ap) {
+    char buf[1024];
+    vsnprintf(buf, sizeof(buf), fmt, ap);
+    fprintf(stdout, "%s: %s\n", tag, buf);
+    return 0;
+}
+
+void* dlopen(const char* filename, int flag) {
+    return (void*)1;
+}
+
+void* dlsym(void* handle, const char* symbol) {
+	// https://github.com/golang/go/issues/59942
+	if (strcmp(symbol, "android_get_device_api_level") != 0) {
+		fprintf(stderr, "Error: Unexpected symbol requested: %s\n", symbol);
+		exit(1);
+	}
+	char buff[PROP_VALUE_MAX];
+	int n = __system_property_get("ro.build.version.sdk", buff);
+	if (n <= 0) {
+		fprintf(stderr, "Error: Failed to get device API level\n");
+		exit(1);
+	}
+	int api_level = atoi(buff);
+	if (api_level < 29) {
+		return NULL;
+	}
+	return (void*)1;
+}
+
+int dlclose(void* handle) {
+    return 0;
+}
+EOF
+
+	${CC:-gcc} -c -o fakes.o fakes.c
+	${AR:-ar} rcs liblog.a fakes.o
```

Comment:
As @twaik pointed out above.
`$CC`, `$AR`, etc. are already setup by the build system so these default fallbacks are both unnecessary and incorrect.

See `termux_setup_toolchain_27b()` for the defaults:
https://github.com/termux/termux-packages/blob/612f58ebe95c84521b261d3113ffb0ce5e1ce22e/scripts/build/toolchain/termux_setup_toolchain_27b.sh#L1-L18

<h1><em></em></h1> <!-- thin separator -->

One more question.
It looks like we're [monkey-patching](https://en.wikipedia.org/wiki/Monkey_patch) the, `__android_log_vprint`, `dlopen`, `dlsym` and `dlclose` functions.

I'd just like to get a couple more details on the thought process behind why you chose to do it this way.

Reply:
golang do not support static build on android as described here https://github.com/golang/go/issues/59942. This was done to make linker happy during static linking.